library/psa_crypto.c

```diff
@@ -1480,20 +1480,79 @@ static void psa_fail_key_creation( psa_key_slot_t *slot )
     psa_wipe_key_slot( slot );
 }
 
+static psa_status_t psa_check_key_slot_attributes(
+    const psa_key_slot_t *slot,
+    const psa_key_attributes_t *attributes )
+{
+    if( attributes->type != 0 )
+    {
+        if( attributes->type != slot->type )
+            return( PSA_ERROR_INVALID_ARGUMENT );
+    }
+
+    if( attributes->domain_parameters_size != 0 )
+    {
+#if defined(MBEDTLS_RSA_C)
+        if( PSA_KEY_TYPE_IS_RSA( slot->type ) )
+        {
+            mbedtls_mpi actual, required;
+            int ret;
+            mbedtls_mpi_init( &actual );
```

Comment:
In a real implementation, we might want to comment that this is comparing the public exponent (the only RSA domain parameter).